web/src/engine/websites/MangaReaderTo.ts

```diff
@@ -0,0 +1,282 @@
+import { Tags } from '../Tags';
+import icon from './MangaReaderTo.webp';
+import type { Chapter } from '../providers/MangaPlugin';
+import { DecoratableMangaScraper, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import DeScramble from '../transformers/ImageDescrambler';
+
+type AJAXResponse = {
+    status: boolean,
+    html: string
+}
+
+type TSlice = {
+    x: number,
+    y: number,
+    width: number,
+    height: number
+}
+
+type TGroup = {
+    slices: number,
+    cols: number,
+    rows: number,
+    x: number,
+    y: number
+}
+
+function ChapterExtractor(element: HTMLElement) {
+    const link = element instanceof HTMLAnchorElement ? (element as HTMLAnchorElement).pathname : element.querySelector<HTMLAnchorElement>('a').pathname;
+    const title = element instanceof HTMLAnchorElement ? element.title.replace(/([^:]*):(.*)/, (match, g1, g2) => g1.trim().toLowerCase() === g2.trim().toLowerCase() ? g1 : match).trim() : element.textContent.trim();
+    const lang = link.match(/\/read\/[^/]+\/([^/]+)/)[1];
+    return {
+        id: link,
+        title: `${title} (${lang})`,
+    };
+
+}
+
+@Common.MangaCSS(/^{origin}\/[^/]+-\d+$/, 'div#ani_detail div.anisc-detail h2.manga-name')
+@Common.MangasMultiPageCSS('/az-list?page={page}', '#main-content div.manga-detail h3 a', 1, 1, 0, Common.AnchorInfoExtractor(true))
+@Common.ChaptersSinglePageCSS('div.chapters-list-ul ul li a, div.volume-list-ul div.manga-poster', ChapterExtractor)
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangareaderto', `MangaReader.to`, 'https://mangareader.to', Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.Multilingual, Tags.Source.Aggregator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const chapterurl = new URL(chapter.Identifier, this.URI);
+        const [data] = await FetchCSS(new Request(chapterurl), 'div#wrapper');
+        const readingId = data.dataset.readingId;
+
+        const uri = new URL(`ajax/image/list/${chapterurl.href.includes('chapter') ? 'chap' : 'vol'}/${readingId}?quality=high`, this.URI);
+        const { status, html } = await FetchJSON<AJAXResponse>(new Request(uri));
+        if (!status) return [];
+
+        const dom = new DOMParser().parseFromString(html, 'text/html');
+        const imagesArr = Array.from(dom.querySelectorAll<HTMLDivElement>('div.iv-card'));
+        return imagesArr.map(image => new Page(this, chapter, new URL(image.dataset.url, this.URI), { shuffled: [...image.classList].includes('shuffled') }));
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
```

Comment:
The complexity of this method is t high, consider extracting parts of it into separate functions.
It is also recommend to place functions which are copied directly from the website script outside of the scraper class.